packages/tspd/test/emitters/markdown/model.test.ts

```diff
@@ -0,0 +1,205 @@
+        "",
+        "",
+        "",
+        "```typespec",
+        "model Lib.Test",
+        "```",
+        "",
+        "## Examples",
+        "",
+        "```tsp",
+        "model Foo {test: Test}",
+        "```",
+        "",
+        "## Properties",
+        "| Name | Type | Description |",
+        "|------|------|-------------|",
+      ].join("\n")
+    );
+  });
+
+  it("render named example", async () => {
+    const result = await renderModel(`
+    /**
+     * @example First Example
+     * \`\`\`tsp
+     * model Foo {test1: Test}
+     * \`\`\`
+     * @example Second Example
+     * \`\`\`tsp
+     * model Foo {test2: Test}
+     * \`\`\`
+     */
+    model Test {}
+  `);
+    expect(result).toEqual(
+      [
+        "# `Test`",
+        "",
+        "",
+        "",
+        "```typespec",
+        "model Lib.Test",
+        "```",
+        "",
+        "## Examples",
+        "### First Example",
+        "",
+        "```tsp",
+        "model Foo {test1: Test}",
+        "```",
+        "",
+        "### Second Example",
+        "",
+        "```tsp",
+        "model Foo {test2: Test}",
+        "```",
+        "",
+        "## Properties",
+        "| Name | Type | Description |",
+        "|------|------|-------------|",
+      ].join("\n")
+    );
+  });
+});
+
+describe("properties table", () => {
+  const common = [
+    "# `Test`",
+    "",
+    "",
+    "```typespec",
+    "model Lib.Test",
+    "```",
+    "",
+    "",
+    "## Properties",
+    "| Name | Type | Description |",
+    "|------|------|-------------|",
+  ];
+  async function expectTable({ code, rows }: { code: string; rows: string[] }) {
+    const result = await renderModel(code);
+    expect(result).toEqual([...common, ...rows].join("\n"));
+  }
+  it("render properties without docs", async () => {
+    await expectTable({
+      code: `model Test { name: string, other: int32 }`,
+      rows: ["| name | `string` |  |", "| other | `int32` |  |"],
+    });
+  });
+
+  it("render properties with documentation", async () => {
+    await expectTable({
+      code: `model Test { 
+        /** name of the test */
+        name: string;
+        /** other of the test */
+        other: int32;
+      }`,
+      rows: ["| name | `string` | name of the test |", "| other | `int32` | other of the test |"],
+    });
+  });
+
+  it("render optional properties", async () => {
+    await expectTable({
+      code: `model Test { name?: string, other: int32 }`,
+      rows: ["| name? | `string` |  |", "| other | `int32` |  |"],
+    });
+  });
+
+  it("render link to local type", async () => {
+    await expectTable({
+      code: `model Test { other: Other } model Other {name: string}`,
+      rows: ["| other | [`Other`](#other) |  |"],
+    });
+  });
+
+  it("render indexer", async () => {
+    await expectTable({
+      code: `model Test is Record<string> { name: string }`,
+      rows: ["| name | `string` |  |", "|  | `string` | Additional properties |"],
```

Comment:
This seems weird. Why would the type not just be `Record<string>`?  It's definitely not a plain string, right?

Reply:
it is saying here that there is additional properties that are string